robot-server/tests/integration/http_api/persistence/test_compatibility.py

```diff
@@ -80,7 +135,11 @@ async def test_protocols_analyses_and_runs_available_from_older_persistence_dir(
                         page_length=999999,  # Big enough to include all commands.
                     )
                 ).json()
-                assert len(all_command_summaries["data"]) > 0
+
+                if len(all_command_summaries["data"]) == 0:
+                    assert run_id in snapshot.runs_with_no_commands
+                else:
+                    assert len(all_command_summaries["data"]) > 0
```

Comment:
I think this is behaviorally equivalent, but I'd find this clearer if we wrote it this like this:

```suggestion

                if run_id in snapshot.runs_with_no_commands:
                    assert len(all_command_summaries["data"]) == 0
                else:
                    assert len(all_command_summaries["data"]) > 0
```

So that the expected behavior of a `Snapshot` depends on the `Snapshot`'s configuration, instead of the other way around.

Reply:
fixed